src/service.rs

```diff
@@ -366,8 +474,46 @@ impl Service {
                                 }
                             }
                         }
+                        HandlerOut::RequestTimedOut(request_id, node_id) => {
+                            if CHECK_VERSION(&self.local_enr.read(), vec![NAT]) {
+                                // Drop the request and attempt establishing the connection to the peer via the
+                                // NAT traversal protocol for sending future requests to the peer, if this peer
+                                // was forwarded to us in a NODES response and we hence have a relay for it.
+                                if self.active_requests.get(&request_id).is_some() {
+                                    let local_enr = self.local_enr.read().clone();
+                                    if let Some(relays) = self.relays.get(&node_id) {
+                                        if let Some(contact_relay) = relays.iter().next() {
+                                            trace!("Requests to peer, that we have learnt of in some NODES response, keep timing out. Peer may be behind an asymmetric NAT. Trying to connect to peer via the NAT traversal protocol. Peer: {}, Relay: {}", node_id, contact_relay);
```

Comment:
While logging this is a bit too long. You could leave out the first part as a comment and as log just the end
```suggestion
                                            trace!("Trying to connect to peer via relay. Peer: {node_id}, Relay: {contact_relay}");
```